### PR TITLE
Add Support for Swift 5.7

### PR DESCRIPTION
**Pull request by Andrea-Scuderi**
# Add support for Swift 5.7

- Add new action signatures with async/await and throws
- Backward compatibility with old action signatures

### swift57Action
- Add support async to Whisk class

### swiftbuild.py
- simplify the code added by the build process to:
```swift
try await _WhiskRuntime.wiskRunLoop { jsonData in
   await _WhiskRuntime.runAsyncMain(mainFunction: <action>, json: jsonData)
}
```

### swiftbuild.py.launcher.swift
- Add `_WhiskRuntime` struct 
- Refactor Error messages
- Add new signatures

```swift
// Examples of Actions supported by Swift 5.7

// Action with Any Input and Any Output
func main(args: Any) -> Any {
    let newArgs = args as! [String:Any]
    if let name = newArgs["name"] as? String {
        return [ "greeting" : "Hello \(name)!" ]
    } else {
        return [ "greeting" : "Hello stranger!" ]
    }
}
 
// Async Action with Any Input and Any Output
func mainAsync(args: Any) async -> Any {
    do {
        //async code sleep for 1 sec
        try await Task.sleep(nanoseconds: 1_000_000_000)
        
        let newArgs = args as! [String:Any]
        if let name = newArgs["name"] as? String {
            return [ "greeting" : "Hello \(name)!" ]
        } else {
            return [ "greeting" : "Hello stranger!" ]
        }
    } catch {
        return ["error" : error.localizedDescription]
    }
}

// Async throwing Action with Any Input and Any Output
func mainAsyncThrows(args: Any) async throws -> Any {
    //async code sleep for 1 sec
    try await Task.sleep(nanoseconds: 1_000_000_000)
    
    let newArgs = args as! [String:Any]
    if let name = newArgs["name"] as? String {
        return [ "greeting" : "Hello \(name)!" ]
    } else {
        return [ "greeting" : "Hello stranger!" ]
    }
}

struct Input: Codable {
    let name: String?
}

struct Output: Codable {
    let count: Int
}

// Action with Codable Input and completion with Codable Output and Error
func mainCompletionCodable(input: Input, completion: @escaping (Output?, Error?) -> Void) -> Void {
    if let name = input.name {
        let output = Output(count: name.count)
        completion(output, nil)
    } else {
        let output = Output(count: 0)
        completion(output, nil)
    }
}

// Action with Codable Input and completion with Codable Output and Error
func mainCompletionCodableNoInput(completion: @escaping (Output?, Error?) -> Void) -> Void {
    let output = Output(count: 0)
    completion(output, nil)
}

// Async throwing Action with Codable Output
func mainCodableAsyncThrowsNoInput() async throws -> Output? {
    try await Task.sleep(nanoseconds: 1_000_000_000)
    return Output(count: 0)
}

// Async throwing Action with a Codable Input and a Codable Output
func mainCodableAsyncThrows(input: Input) async throws -> Output? {
    try await Task.sleep(nanoseconds: 1_000_000_000)
    if let name = input.name {
        return Output(count: name.count)
    } else {
        return Output(count: 0)
    }
}
```

Merry Christmas and Happy New Year! 🎄

### CODE REVIEW

**Comment by rabbah** [comment]
Thanks @Andrea-Scuderi for this contribution. 

Do you have an Apache CLA on file I was not able to locate one.

We may continue to hold merging this PR until we've cut over to the new GHA flow and away from Travis.

**Comment by Andrea-Scuderi** [comment]
> Do you have an Apache CLA on file I was not able to locate one.

Yes, I've sent a CLA the 11/11/2019.

**Comment by dgrove-oss** [comment]
Hi @Andrea-Scuderi  -- we've finally managed to get GitHub Action based CI working for this repository.  Apologies for the very long delay.  Would you have the time and interest to rebase this PR to the current master so we can get it merged?

File: core/swift54Action/CHANGELOG.md [hunk 1]
```diff
@@ -17,7 +17,7 @@
 #
 -->
 
-# Apache OpenWhisk Swift 5.4 Runtime Container
+# Apache OpenWhisk Swift 5.7 Runtime Container
```
Comment (dgrove-oss) [inline core/swift54Action/CHANGELOG.md:20]: Minor nit; this file shouldn't change.
```suggestion
# Apache OpenWhisk Swift 5.4 Runtime Container
```

File: core/swift57Action/CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+-->
+
+# Apache OpenWhisk Swift 5.7 Runtime Container
+
+- Support for swift async/await
+
+## 1.17.0
```
Comment (dgrove-oss) [inline core/swift57Action/CHANGELOG.md:24]: We'll fill in the actual number as part of making the release
```suggestion
## Next Apache Release
```

File: tools/travis/build.sh [hunk 3]
```diff
@@ -49,6 +49,7 @@ TERM=dumb ./gradlew \
 :core:swift51Action:distDocker \
 :core:swift53Action:distDocker \
 :core:swift54Action:distDocker \
+:core:swift57Action:distDocker \
```
Comment (dgrove-oss) [inline tools/travis/build.sh:52]: The corresponding addition should be made after this line now:  https://github.com/apache/openwhisk-runtime-swift/blob/master/.github/workflows/ci.yaml#L103

File: tools/travis/publish.sh [hunk 4]
```diff
@@ -36,6 +36,8 @@ elif [ ${RUNTIME_VERSION} == "5.3" ]; then
   RUNTIME="swift53Action"
 elif [ ${RUNTIME_VERSION} == "5.4" ]; then
   RUNTIME="swift54Action"
+elif [ ${RUNTIME_VERSION} == "5.7" ]; then
```
Comment (dgrove-oss) [inline tools/travis/publish.sh:39]: The corresponding change should be made here now: https://github.com/apache/openwhisk-runtime-swift/blob/master/.github/workflows/ci.yaml#L116

**Review by dgrove-oss — CHANGES_REQUESTED** [review]
If you have the time to accept the two small changes to the CHANGELOG files and update the .github/workflows/ci.yml that would be great.

File: core/swift57Action/CHANGELOG.md (hunk 2 above)
Comment (Andrea-Scuderi) [reply core/swift57Action/CHANGELOG.md:24]: https://github.com/apache/openwhisk-runtime-swift/pull/153/commits/4917c54a630c82b33a68ec0d66139b1657652ebe

File: tools/travis/build.sh (hunk 3 above)
Comment (Andrea-Scuderi) [reply tools/travis/build.sh:52]: https://github.com/apache/openwhisk-runtime-swift/pull/153/commits/0707e20ec1070a6231135bf129de81b56bb91f49

File: tools/travis/publish.sh (hunk 4 above)
Comment (Andrea-Scuderi) [reply tools/travis/publish.sh:39]: https://github.com/apache/openwhisk-runtime-swift/pull/153/commits/0707e20ec1070a6231135bf129de81b56bb91f49

**Comment by Andrea-Scuderi** [comment]
@dgrove-oss I've updated the branch with your suggestions. Could you run the CI?
Thanks

**Comment by dgrove-oss** [comment]
Thanks for the quick response!  CI run approved.

File: core/swift54Action/CHANGELOG.md (hunk 1 above)
Comment (Andrea-Scuderi) [reply core/swift54Action/CHANGELOG.md]: https://github.com/apache/openwhisk-runtime-swift/commit/4917c54a630c82b33a68ec0d66139b1657652ebe

**Review by dgrove-oss — APPROVED** [review]
LGTM -- Thanks for your patience!

**Pull request merged**